pytorch_lightning/trainer/configuration_validator.py

```diff
@@ -90,6 +90,17 @@ def __verify_train_loop_configuration(self, model: "pl.LightningModule") -> None
                 "(rather, they are called on every optimization step)."
             )
 
+        # Allow max_epochs or max_steps to be zero, since this will be handled by fit_loop.done
+        if trainer.max_epochs and (trainer.max_epochs < -1 or isinstance(trainer.max_epochs, float)):
```

Comment:
I don't think we need to check for `float` because the argument is typed as `Optional[int]`

Reply:
@tchaton requested this check + the typing doesn't actually validate the argument. Eg:

```
>>> def test(x: int):
...     print(x)
... 
>>> test(5.0)
5.0
```

doesn't throw any errors even though the type that should be passed is an int.

Reply:
it seems arbitrary to check for float. I could just pass a coconut and you wouldn't catch it, right?

I would say either drop the float check or validate against all incompatible options :))

Reply:
Updated to specifically check for int to catch that 🥥